src/components/Header/Header.jsx

```diff
@@ -39,6 +41,30 @@ function Header() {
         }
     }
 
+    const handleToggleMouseLeave = useCallback(() => {
+        globalDispatch({ type: actionTypes.SET_THEME });
+    }, [globalDispatch]);
+
+    const handleToggleMouseEnter = useCallback(() => {
+        themeToggleRef.current.addEventListener('mouseleave', handleToggleMouseLeave);
+        globalDispatch({ type: actionTypes.SET_THEME });
+    }, [globalDispatch, handleToggleMouseLeave]);
+
+    const handleToggleMouseClick = () => {
+        themeToggleRef.current.removeEventListener('mouseleave', handleToggleMouseLeave);
+    }
```

Comment:
Esta experiencia de usuario me parece un poco confusa; funciona para el primer clic, pero si doy varios antes de quitar el puntero no regresa al estado anterior. También me sacó de onda que se aplicara el tema con el puro hover.
Mi sugerencia sería que los cambios de tema se apliquen hasta que se haga el clic y quizá agregar un tooltip al switch tipo "Switch to light/dark theme" dependiendo del estado de la app. La animación en el ícono me agrada.

Reply:
Entiendo, me pareció una buena idea, pero tienes razón, puede resultar confuso para el usuario. Seguí tu recomendación y agregué un tooltip en lugar de cambiar el tema on hover. Este cambio fue echo en el commit 5a6f2cd